npm/private/transitive_closure.bzl

```diff
@@ -155,12 +155,19 @@ def translate_to_transitive_closure(lock_importers, lock_packages, prod = False,
     package_version_map = {}
 
     for package_path, package_snapshot in lock_packages.items():
-        package, package_info = _gather_package_info(package_path, package_snapshot)
-        packages[package] = package_info
+        package_key, package_info = _gather_package_info(package_path, package_snapshot)
+
+        if package_key in packages:
```

Comment:
This is an error/bug that exists with the lockfile on main right now.

The key `_gather_package_info` produces is not unique for at least 1 case where there are 2 versions of a dependency with the exact same name+version but a different URL.

(Also note I renamed `package` to `package_key` since it is supposed to be a unique key...)